tests/functional/test_install.py

```diff
@@ -627,6 +628,60 @@ def test_install_package_with_target(script):
     assert singlemodule_py in result.files_updated, str(result)
 
 
+def test_install_nonlocal_compatible_wheel(script, data):
+    target_dir = script.scratch_path / 'target'
+
+    # Test install with --target
+    result = script.pip(
+        'install',
+        '-t', target_dir,
+        '--no-index', '--find-links', data.find_links,
+        '--only-binary=:all:',
+        '--python', '3',
+        '--platform', 'fakeplat',
+        '--abi', 'fakeabi',
+        'simplewheel',
+    )
+    distinfo = Path('scratch') / 'target' / 'simplewheel-2.0.dist-info'
+    assert distinfo in result.files_created
+
+    # Test install without --target
+    result = script.pip(
```

Comment:
This result isn't being used in any way. I'd suggest adding a check for the exit code immediately after.

Out the back of my head, I think expect_error lets a test pass even if there's no errors. If that's the case, the above would be needed anyway. :)